src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -680,24 +798,120 @@ def get_observation(self, event: Event) -> pd.DataFrame:
             | (df_eligible_dependents["last_year_income"] < 4_300)
         ]
 
-        s_eligible_dependent_id_list = df_eligible_dependents.groupby("guardian_id").apply(
-            lambda df_g: list(df_g.index)
-        )
+        df = df_eligible_dependents
+        df["dependent_id"] = df.index
+        df["tax_year"] = event.time.year - 1
 
-        s_eligible_dependent_address_id_list = df_eligible_dependents.groupby(
-            "guardian_id"
-        ).apply(lambda df_g: list(df_g["address_id"]))
+        # merge in simulant columns based on simulant id
+        for col in [
+            "first_name_id",
+            "middle_name_id",
+            "last_name_id",
+            "age",
+            "date_of_birth",
+            "sex",
+            "ssn",
+            "address_id",
+            "housing_type",
+        ]:
+            df[col] = df["dependent_id"].map(pop_full[col])
 
-        for col, s in [
-            ["eligible_dependent_id_list", s_eligible_dependent_id_list],
-            ["eligible_dependent_address_id_list", s_eligible_dependent_address_id_list],
+        # delete extraneous columns
+        for col in [
+            "in_united_states",
+            "alive",
+            "guardian_1",
+            "guardian_2",
+            "last_year_income",
+            "race_ethnicity",
+            "housing_type",
+            "tracked",
         ]:
-            df_w2[col] = df_w2["simulant_id"].map(s)
+            del df[col]
 
-        # re-initialize income-to-date series for next year of income counting
-        self.income_to_date = empty_income_series()
+        return df
 
-        return df_w2[self.output_columns]
+
+class Tax1040Observer(BaseObserver):
+    """Class for observing columns relevant to 1040 tax data (that is not
+    already recorded by W2 or dependents observer)
```

Comment:
If I'm understanding this code correctly, I think this comment is misleading. The only extra column observed here is `relation_to_household_head`.

Instead, the way I understand this observer is that it is different in the time dimension: it observes the taxpayers as of April instead of January, so things may have changed in the meantime.

Reply:
> If I'm understanding this code correctly, I think this comment is misleading. The only extra column observed here is relation_to_household_head.
> 
> Instead, the way I understand this observer is that it is different in the time dimension: it observes the taxpayers as of April instead of January, so things may have changed in the meantime.

Good point, I have updated the docstring